### DISCUSSION TITLE
Overwrite baseStyle defined in checkbox component

### DISCUSSION THREAD

**Discussion by pedroabreu** [chakra-ui/chakra-ui#4753] [General]
Hi all,

I have more of a generic question regarding overwriting baseStyles. In the checkbox component, I need to overwrite the `label` component, that has `baseStyles` defined in the `checkbox` component, see https://github.com/chakra-ui/chakra-ui/blob/main/packages/checkbox/src/checkbox.tsx#L29

```
const Label = chakra("label", {
  baseStyle: {
    cursor: "pointer",
    display: "inline-flex",
    alignItems: "center",
    verticalAlign: "top",
    position: "relative",
    _disabled: {
      cursor: "not-allowed",
    },
  },
})
``` 

I can see I 'should' be able to overwrite it by defining the `container` in the theme but because it's set using the `_css` prop it won't overwrite those styles.

```
    <Label
      __css={styles.container}
      className={cx("chakra-checkbox", className)}
      {...getRootProps()}
    >
```

Should these baseStyles be defined in the theme so we can easily overwrite them? Mostly trying to understand the reasoning for them to be defined in the component.

Is there a nicer way to overwrite those styles than setting the `sx` prop in my component? See example bellow

In the theme
```
...
  icon: baseStyleIcon,
  control: baseStyleControl(props),
  label: baseStyleLabel,
  container: {
      alignItems: 'flex-start',
  }
```

and in the component

```
...
<Checkbox sx={styles.container} ...
```

Thanks in advance!

**Comment by primos63** [comment]
> I can see I 'should' be able to overwrite it by defining the container in the theme but because it's set using the _css prop it won't overwrite those styles.

That is the usual prop for styles from the theme. The theme style has the lowest precedence (__css, baseStyle, styleProps and sx).
This can be seen in the [toCSSObject](https://github.com/chakra-ui/chakra-ui/blob/4e2d7b5899467a74df81f2ed601424fd726980e7/packages/system/src/system.ts#L44) which produces the CSS styles
```javascript
  const styleProps = objectFilter(rest, (_, prop) => isStyleProp(prop))
  const finalBaseStyle = runIfFn(baseStyle, props)
  const finalStyles = Object.assign(
    {},
    __css,
    finalBaseStyle,
    filterUndefined(styleProps),
    sx,
  )
```
The `baseStyle` comes from the styling of the element. In this case, 
```jsx
const Label = chakra("label", {
  baseStyle: {
    cursor: "pointer",
    display: "inline-flex",
    alignItems: "center",
    verticalAlign: "top",
    position: "relative",
    _disabled: {
      cursor: "not-allowed",
    },
  },
})
```
Other elements styled in a similar fashion can be found in `input-addon.tsx`, `input-element.tsx`, `center.tsx`, `spacer.tsx`, `number-input.tsx`, `circular-progress.tsx`, `select.tsx` and `skeleton.tsx`.

> Should these baseStyles be defined in the theme so we can easily overwrite them? Mostly trying to understand the reasoning for them to be defined in the component.

Probably not. Chakra is not completely unopinionated. Note that in the case of the checkbox there is no part for `container` in the checkbox theme. That may suggest it's not something for public access or simply no other overridable styles were needed. Note that the checkbox `control` part has a `baseStyle` defined in the code and overridable properties in the theme and there is no intersection between them. All that said, even though Chakra has some opinions on styling they allow you to override it via the `sx` prop.

> Is there a nicer way to overwrite those styles than setting the sx prop in my component?

If you're using checkbox in several places and want to avoid setting the `sx` prop for each, you may want to create a wrapper component for checkbox which sets the `sx` prop to `styles.container`.
